packages/core/src/components/Dialog/Dialog.jsx

```diff
@@ -0,0 +1,120 @@
+import AriaModal from 'react-aria-modal';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export const Dialog = function(props) {
+  const {
+    actions,
+    ariaCloseLabel,
+    children,
+    escapeExitDisabled,
+    onExit,
+    title,
+    ...modalProps
+  } = props;
+
+  return (
+    <AriaModal
+      dialogClass="ds-c-dialog"
+      escapeExits={!escapeExitDisabled}
+      includeDefaultStyles={false}
+      onExit={onExit}
+      titleId="dialog-title"
+      underlayClass="ds-c-dialog-wrap"
+      {...modalProps}
+    >
+      <div role="document">
+        <header className="ds-c-dialog__header" role="banner">
+          {title && (
+            <h1 className="ds-h2" id="dialog-title">
+              {title}
+            </h1>
+          )}
+          <button
+            aria-label={ariaCloseLabel}
+            className="ds-c-button ds-c-button--transparent ds-c-dialog__close"
+            onClick={onExit}
+          >
+            Close
+          </button>
+        </header>
+        <main role="main">{children}</main>
+        {actions && (
+          <footer className="ds-c-dialog__actions" role="contentinfo">
```

Comment:
> I think these action buttons could be wrapped in a `<footer role="contentinfo"` tag, so the modal has 3 distinct landmark regions.

@1Copenut  Should this be `contentinfo` or `complementary`? These would be primary actions related to the `main` content. For example, `main` might include form fields and the actions might include a submit button.

Reply:
You raise a good point. `aside role="complementary"` feels better. After a bit more digging, this Treehouse article supports your thinking: http://blog.teamtreehouse.com/use-html5-sectioning-elements

```
The <footer> element is used to represent information about a section such as the author, copyright information, or links to related web pages.
```

Let's change it to an `aside` with role `complementary`. Good call.